File: [프로그래머스] 연습문제_대충 만든 자판.js
function solution(keymap, targets) {
    var answer = Array(targets.length).fill(0);
    let dic = {};
    for(let i of keymap){
        for(let j of i){
            if(j in dic){
                dic[j] = Math.min(dic[j], i.indexOf(j) + 1);
            } else {
                dic[j] = i.indexOf(j) + 1;
            }
        }
    }
    for(let i = 0; i < targets.length; i++){
        for(let j of targets[i]){
            if(j in dic){
                answer[i] += dic[j]
            } else {
                answer[i] = -1; 
                break;
            }
        }
    }
    return answer;
}

File: [프로그래머스] 연습문제_대충 만든 자판.py
def solution(keymap, targets):
    answer = [0] * len(targets)
    dic = {}
    for i in keymap:
        for j in i:
            if j in dic:
                if dic[j] > i.index(j)+1:
                    dic[j] = i.index(j)+1
            else:
                dic[j] = i.index(j)+1
    for i in range(len(targets)):
        for j in targets[i]:
            if j in dic:
                answer[i] += dic[j]
            else:
                answer[i] = -1
                break
    return answer